python/ray/data/datasource/image_folder_datasource.py

```diff
@@ -61,70 +60,60 @@ class ImageFolderDatasource(BinaryDatasource):
-        >>>
         >>> ds = oe.fit_transform(ds)  # doctest: +SKIP
-        >>>
         >>> sample = ds.take(1)[0]  # doctest: +SKIP
         >>> sample["label"]  # doctest: +SKIP
         71
     """  # noqa: E501
 
     def create_reader(
         self,
-        paths: Union[str, List[str]],
-        filesystem: Optional["pyarrow.fs.FileSystem"] = None,
-        partition_filter: PathPartitionFilter = None,
-        **kwargs,
+        root: str,
+        size: Tuple[int, int],
     ) -> "Reader[T]":
-        if len(paths) != 1:
-            raise ValueError(
-                "`ImageFolderDatasource` expects 1 path representing the dataset "
-                f"root, but it got {len(paths)} paths instead. To fix this "
-                "error, pass in a single-element list containing the dataset root "
-                '(for example, `paths=["s3://imagenet/train"]`)'
-            )
+        """Return a :py:class:`Reader` that reads images.
 
-        try:
-            import imageio  # noqa: F401
-        except ImportError:
-            raise ImportError(
-                "`ImageFolderDatasource` depends on 'imageio', but 'imageio' couldn't "
-                "be imported. You can install 'imageio' by running "
-                "`pip install imageio`."
-            )
-
-        if partition_filter is None:
-            partition_filter = FileExtensionFilter(file_extensions=IMAGE_EXTENSIONS)
+        Args:
+            root: Path to the dataset root.
+            size: The desired height and width of loaded images.
+        """
+        self._check_import(module="imageio", package="imagio")
+        self._check_import(module="skimage", package="scikit-image")
 
         # We call `_resolve_paths_and_filesystem` so that the dataset root is formatted
         # in the same way as the paths passed to `_get_class_from_path`.
-        paths, filesystem = _resolve_paths_and_filesystem(paths, filesystem)
-        self.root = paths[0]
+        paths, _ = _resolve_paths_and_filesystem([root])
+        root = paths[0]
 
         return super().create_reader(
             paths=paths,
-            filesystem=filesystem,
-            partition_filter=partition_filter,
-            **kwargs,
+            partition_filter=FileExtensionFilter(file_extensions=IMAGE_EXTENSIONS),
+            root=root,
+            size=size,
         )
 
-    def _read_file(self, f: "pyarrow.NativeFile", path: str, **reader_args):
+    def _read_file(
+        self, f: "pyarrow.NativeFile", path: str, root: str, size: Tuple[int, int]
+    ):
         import imageio as iio
         import pandas as pd
         from ray.data.extensions import TensorArray
+        import skimage
 
         records = super()._read_file(f, path, include_paths=True)
         assert len(records) == 1
         path, data = records[0]
 
         image = iio.imread(data)
-        label = _get_class_from_path(path, self.root)
+        image = skimage.transform.resize(image, size)
```

Comment:
Is this the most standard library for resizing images? Also, should we give options to customize the resizing process?

Reply:
> Is this the most standard library for resizing images?

I don't know if there's a de-facto standard. I chose scikit-image because it was easiest.

The three options are:
* PIL
* OpenCV
* scikit-image

PIL is clunky. To resize the array with PIL, we'd have to convert the array to a PIL image and then back to a NumPy array.
```
img = Image.fromarray(arr)
img.resize(size=(100, 100))
arr = np.array(img)
```

OpenCV is too big. According to Wikapedia, OpenCV is ~200MB.

Reply:
> Also, should we give options to customize the resizing process?

I think we should wait until someone asks.

For context, we could give any of the options exposed by scikit-image.

```
skimage.transform.resize(image, output_shape, order=None, mode='reflect', cval=0, clip=True, preserve_range=False, anti_aliasing=None, anti_aliasing_sigma=None)
```

TorchVision only exposes interpolation mode and anti-aliasing.

```
torchvision.transforms.Resize(size, interpolation=InterpolationMode.BILINEAR, max_size=None, antialias=None)
```

Reply:
Got it. Seems like we should revisit later with ray.data.read_images(), but this seems sensible for now.